jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/bpmn2/ProcessDescriptor.java

```diff
@@ -177,6 +181,22 @@ public void setSignals( Collection<String> signals ) {
        this.signals = signals;
     }
 
+    public Collection<Signal> getSignalsMetadata() {
+        return signalsMetadata;
```

Comment:
it is not signal metadata... it is the signal element it self. change for a more descriptive name. i.e signallement or signalobject.